crates/flux-syntax/src/surface.rs

```diff
@@ -153,14 +153,22 @@ pub struct FnSig<R = ()> {
     pub args: Vec<Arg<R>>,
     /// example `i32{v:v >= 0}`
     pub returns: FnRetTy<R>,
-    /// example: `*x: i32{v. v = n+1}`
-    pub ensures: Vec<(Ident, Ty<R>)>,
+    /// example: `*x: i32{v. v = n+1}` or just `x > 10`
+    pub ensures: Ensures<R>,
```

Comment:
Any reason to not make this a list of constraints (where a constraint is either `x: T` or `pred`) like in the lower levels?

Reply:
duh. let me do that!